enterprise/frontend/src/embedding-sdk/lib/theme/derived-colors.ts

```diff
@@ -0,0 +1,93 @@
+import { applyColorOperation } from "metabase/embedding-sdk/theme/dynamic-css-vars";
+import { DYNAMIC_SDK_DEFAULT_COLORS_CONFIG } from "metabase/embedding-sdk/theme/dynamic-sdk-color-defaults";
+import type { MappableSdkColor } from "metabase/embedding-sdk/theme/embedding-color-palette";
+import { SDK_TO_MAIN_APP_COLORS_MAPPING } from "metabase/embedding-sdk/theme/embedding-color-palette";
+import type { SourceColorKey } from "metabase/embedding-sdk/types/private/css-variables";
+import { colors, isDark } from "metabase/lib/colors";
+import type { ColorName, ColorPalette } from "metabase/lib/colors/types";
+import type { MantineThemeOverride } from "metabase/ui";
+
+import { colorTuple } from "./color-tuple";
+
+/**
+ * Generate derived colors for SDK colors that the user did not define.
+ * Important: the `override` must be pre-populated with mapped colors.
+ *
+ * Returns a Mantine theme override with the derived colors already in colorTuple format.
+ */
+export function getDerivedSdkDefaultColors({
+  override,
+  appColors,
+}: {
+  override: MantineThemeOverride;
+  appColors?: ColorPalette;
+}): MantineThemeOverride {
+  // Resolve colors that are used as the base for deriving colors.
+  // Example: we might derive `background-hover` from `bg-white`
+  const getColor = (colorName: SourceColorKey): string => {
+    // If the color has been defined in the Mantine theme override, use it.
+    if (override.colors?.[colorName]) {
+      const tuple = override.colors[colorName];
+
+      if (Array.isArray(tuple) && tuple.length > 0) {
+        return tuple[0];
+      }
```

Comment:
In Metabase, the entire color tuple is the same color as we do not use the shading system.